lib/backends/webgl/op-resolve-rules.ts

```diff
@@ -47,6 +49,7 @@ export const WEBGL_OP_RESOLVE_RULES: ReadonlyArray<OpSet.ResolveRule> = [
   ['Clip', '', '6-10', () => new WebGLClip()],
   ['Concat', '', '4+', () => new WebGLConcat()],
   ['Conv', '', '1+', () => new WebGLConv()],
+  ['ConvPacked', '', '1+', () => new WebGLConvPacked()],
```

Comment:
>ConvPacked [](start = 4, length = 10)

we need to find a good way to resolve kernels with both pack and unpack version supported. One way is through webGL session handler's resolve method, another is through individual kernels. 

The one through session handler is more systematic, but as it calls into a generic `resolveOperator` function, which is shared between all backends, it makes it harder to make any webGL backend specific changes. One possible solution is to change `resolveOperator` to take a function pointer instead, and have each backends decide how to resolve ops specifically. However, this would require a lot of change. The other option is more light-weighted. Similar to `concat`, it queries session handler and append different source code for fragment shader. 

Personally I feel option one should be the goal for long run as it is more centralized and clean, but may have a lower priority given current planning. I'm also good with option 2 for now too.

Reply:
Same for MatmulPack below.

---
In reply to: [596286236](https://github.com/microsoft/onnxjs/pull/274#discussion_r596286236) [](ancestors = 596286236)

Reply:
right. let's have a discussion offline.

Reply:
Fixed. Now it's aligned with concat and reshape pack version.